lib/matplotlib/tight_layout.py

```diff
@@ -171,25 +171,46 @@ def auto_adjust_subplotpars(
         margin_bottom = max([sum(s) for s in vspaces[-cols:]] + [0])
         margin_bottom += pad_inches / fig_height_inch
 
+    if margin_left + margin_right >= 1:
+        margin_left = 0.4999
+        margin_right = 0.4999
+        warnings.warn('The left and right margins cannot be made large '
+                      'enough to accomodate all axes decorations. ')
+    if margin_bottom + margin_top >= 1:
+        margin_bottom = 0.4999
+        margin_top = 0.4999
+        warnings.warn('The bottom and top margins cannot be made large '
+                      'enough to accomodate all axes decorations. ')
+
     kwargs = dict(left=margin_left,
                   right=1 - margin_right,
                   bottom=margin_bottom,
                   top=1 - margin_top)
-
     if cols > 1:
         hspace = (
             max(sum(s)
                 for i in range(rows)
                 for s in hspaces[i * (cols + 1) + 1:(i + 1) * (cols + 1) - 1])
             + hpad_inches / fig_width_inch)
+        # axes widths:
         h_axes = (1 - margin_right - margin_left - hspace * (cols - 1)) / cols
-        kwargs["wspace"] = hspace / h_axes
+        if h_axes < 0:
+            warnings.warn('tight_layout cannot make axes width small enough '
+                          'to accomodate all axes decorations')
+            kwargs["wspace"] = 0.5
+        else:
+            kwargs["wspace"] = hspace / h_axes
 
     if rows > 1:
         vspace = (max(sum(s) for s in vspaces[cols:-cols])
                   + vpad_inches / fig_height_inch)
         v_axes = (1 - margin_top - margin_bottom - vspace * (rows - 1)) / rows
-        kwargs["hspace"] = vspace / v_axes
+        if v_axes < 0:
+            warnings.warn('tight_layout cannot make axes height small enough '
+                          'to accomodate all axes decorations')
+            kwargs["hspace"] = 0.01
```

Comment:
This and line 200 above fall back to different numbers, which was intended?

Reply:
Thanks!  Fixed.